test/e2e/metering_manual_install_tests.go

```diff
@@ -107,6 +108,141 @@ func testManualMeteringInstall(
 }
 
+func createNFSProvisioner(ctx *deployframework.DeployerCtx) error {
+	// create namespace
+	ctx.TargetPodsCount = nonHDFSTargetPodCount
+
+	_, err := ctx.Client.CoreV1().Namespaces().Get(context.Background(), ctx.Namespace, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil
+	}
+	if apierrors.IsNotFound(err) {
+		n := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ctx.Namespace,
+				Labels: map[string]string{
+					"name": ctx.Namespace + "-" + testingNamespaceLabel,
+				},
+			},
+		}
+		_, err = ctx.Client.CoreV1().Namespaces().Create(context.Background(), n, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		ctx.Logger.Debugf("Created the %s namespace", ctx.Namespace)
+	}
+
+	//decode and create yamls except for persistentvolume
+
+	type nfsManifestItem struct {
+		resourceType interface{}
+		resourceFile string
+		actions      []string
+	}
+	type nfsServer struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+
+	type nfsService struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+
+	type nfsStorageclass struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+	var nfsManifestItems = map[string]nfsManifestItem{
+		"nfsServer":
+		{
+
+			resourceType: nfsServer{},
+			resourceFile: "/test/testhelpers/nfs_server.yaml",
+			actions:      []string{"CREATE"},
+		},
+		"nfsService":
+		{
+			resourceType: nfsService{},
+			resourceFile: "/test/testhelpers/nfs_service.yaml",
+			actions:      []string{"CREATE", "GET"},
+		},
+		"nfsStorageclass":
+		{
+			resourceType: nfsStorageclass{},
+			resourceFile: "/test/testhelpers/nfs_storageclass.yaml",
+			actions:      []string{"CREATE"},
+		},
+	}
+
+	for _, manifest := range nfsManifestItems {
+
+		err := deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsServer"].resourceFile, nfsManifestItems["nfsServer"].resourceType)
+		err := deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsService"].resourceType)
+		err := deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsStorageclass"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceType)
+		if err != nil {
+			return fmt.Errorf("failed to decode the %s YAML service manifest: %v", err, manifest.resourceFile)
+		}
+
+		for _, action := range manifest.actions {
+			if action == "CREATE" {
+				ctx.Client.CoreV1().Namespaces().Create(context.Background(), &nfsManifestItems[0].resourceType, metav1.CreateOptions{})
+			}
+			ctx.Logger.Debugf("Created the %s nfs_server %s nfs_service %s nfs_storageclass", nfsManifestItems["nfsServer"],nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceFile)
+			if action == "GET" {
+
+				foo, err := ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.Background(), "nfs-service", metav1.GetOptions{})
+				if err != nil && !apierrors.IsNotFound(err) {
+					return nil
+				}
+			}
+		}
+	}
```

Comment:
I would highly suggest we shy away from doing this kind of pattern and making this more complex than it needs to be. Currently, this is pretty difficult to read, and we can cut down on the amount of code dedicated to creating these resources by doing something like the following:

```go
	var (
		deployment *appsv1.Deployment
		service *corev1.Service
		storageclass *storagev1.StorageClass
	)
	resources := map[string]interface{}{
		"testhelpers/manifests/nfs/deployment.yaml": deployment,
		"testhelpers/manifests/nfs/service.yaml": service,
		"testhelpers/manifests/nfs/storageclass.yaml": storageclass, 
	}
	
	for path, resource := range resources {
		/*
		Steps:
		1. GET the resource
		   a. if the resource exists, continue to next item
		   b. else, decode the @path YAML manifest to the @resource object
		2. Create the @resource
	}
...
```